### PR TITLE
Update to protocol 1285

### PR DESCRIPTION
**Pull request by Limmek**
Updated to work with the latest tfs.

### CODE REVIEW

File: OTLoginServer/Classes/WebService.cs [hunk 1]
```diff
@@ -99,8 +101,8 @@ public async Task ProcessRequest(JObject obj, HttpListenerContext ctx)
                     {
                         Session = new Session
                         {
-                            SessionKey = $"{email}\n{password}",
-                            LastLoginTime = account.LastLoginTime,
+                            //SessionKey = $"{email}\n{password}",
```
Comment (nekiro) [inline OTLoginServer/Classes/WebService.cs]: delete this line

File: OTLoginServer/Models/World.cs [hunk 2]
```diff
@@ -4,15 +4,18 @@ public class World
     {
         public int Id { get; set; } = 0;
         public string Name { get; set; } = "World";
+        public string ExternalAddress { get; set; }
+        public int ExternalPort { get; set; }
```
Comment (nekiro) [inline OTLoginServer/Models/World.cs]: what's this?
there is ExternalAddressUnprotected and ExternalPortUnprotected  for that

File: OTLoginServer/Classes/WebService.cs [hunk 3]
```diff
@@ -111,7 +113,6 @@ public async Task ProcessRequest(JObject obj, HttpListenerContext ctx)
                             Worlds = await Task.Run(() => _db.GetWorlds())
                         }
                     };
-
```
Comment (nekiro) [inline OTLoginServer/Classes/WebService.cs]: why?

File: OTLoginServer/Classes/DatabaseManager.cs [hunk 4]
```diff
@@ -78,8 +77,12 @@ public ICollection<World> GetWorlds()
                 Name = ConfigLoader.GetString("serverName"),
             };
 
+            world.ExternalAddress = world.ExternalAddressUnprotected = ConfigLoader.GetString("ip");
```
Comment (nekiro) [inline OTLoginServer/Classes/DatabaseManager.cs]: you just duplicated what ExternalAddressProtected and ExternalAddressUnprotected  is doing

File: OTLoginServer/Classes/DatabaseManager.cs [hunk 5]
```diff
@@ -58,9 +58,8 @@ public async Task<Account> GetAccount(string email, string password)
 
                     account.Id = dataReader.GetInt32(dataReader.GetOrdinal("id"));
 
-                    int premDays = dataReader.GetInt32(dataReader.GetOrdinal("premdays"));
+                    int premDays = dataReader.GetInt32(dataReader.GetOrdinal("premium_ends_at"));
                     account.IsPremium = premDays > 0;
-                    account.LastLoginTime = dataReader.GetInt64(dataReader.GetOrdinal("lastday"));
```
Comment (nekiro) [inline OTLoginServer/Classes/DatabaseManager.cs:63]: is this not used or why delete?

**Review by nekiro — CHANGES_REQUESTED** [review]

File: OTLoginServer/Models/World.cs (hunk 2 above)
Comment (Limmek) [reply OTLoginServer/Models/World.cs]: I was stuck on character loading so i first thouth this was the issue. But it was not related to it at all.

File: OTLoginServer/Classes/WebService.cs (hunk 3 above)
Comment (Limmek) [reply OTLoginServer/Classes/WebService.cs]: idk lol

File: OTLoginServer/Classes/DatabaseManager.cs (hunk 5 above)
Comment (Limmek) [reply OTLoginServer/Classes/DatabaseManager.cs:63]: to get the last login time you need to get it from the character table not from the account.

File: OTLoginServer/Classes/DatabaseManager.cs (hunk 4 above)
Comment (Limmek) [reply OTLoginServer/Classes/DatabaseManager.cs]: Removed my connecting issue was not related to this..

**Review by nekiro — APPROVED** [review]
Looks good, thanks

Fucked up the merge though, rip.

**Comment by Limmek** [comment]
haha! No problem :)

**Pull request merged**